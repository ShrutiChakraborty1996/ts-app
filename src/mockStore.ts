import thunk from "redux-thunk";
import configureStore, {MockStore} from "redux-mock-store";
import TodoType from "./types/TodoType";

export const generateMockStore = (state: TodoType[] | {}): MockStore<TodoType[] | {}> => {
    const mockStore = configureStore([thunk]);
    const store = mockStore(state);
    store.clearActions();
    return store;
};