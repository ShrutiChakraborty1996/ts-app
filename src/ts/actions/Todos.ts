import {ActionTypes} from "../types/ActionType";
import TodoType from "../types/TodoType";
import {ThemeAction} from "./ThemeAction";

export interface AddTodoAction {
    type: ActionTypes.ADD_TODO,
    payload: { todo: TodoType }
}

export function addTodo(title: string): AddTodoAction {
    return {
        type: ActionTypes.ADD_TODO,
        payload: {
            todo: {title}
        }
    }
}

export type Action = AddTodoAction | ThemeAction;