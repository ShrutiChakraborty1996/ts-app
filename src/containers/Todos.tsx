import {connect} from "react-redux";
import Todo from "../components/todo/Todo";
import {addTodo} from "../actions/Todos";
import TodoType from "../types/TodoType";

export interface State {
    todos: TodoType[]
};

const mapStateToProps = (state: State) => {
    return {
        todos: state.todos
    }
};

const mapDispatchToProps = {
    addTodo
};

export default connect(mapStateToProps, mapDispatchToProps)(Todo);