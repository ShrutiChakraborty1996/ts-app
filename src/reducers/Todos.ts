import TodoType from "../types/TodoType";
import {Action} from "../actions/Todos";
import {ActionTypes} from "../types/TodoActionType";

export default (state: TodoType[] = [], action: Action) => {
    switch (action.type) {
        case ActionTypes.ADD_TODO:
            return [...state, action.payload.todo];
        default:
            return state
    }
}

