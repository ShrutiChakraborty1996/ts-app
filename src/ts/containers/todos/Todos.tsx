import {connect} from "react-redux";
import Todo from "../../components/todo/Todo";
import {addTodo} from "../../actions/Todos";
import TodoType from "../../types/TodoType";

export interface TState {
    todos: TodoType[]
}

const mapStateToProps = (state: TState) => {
    return {
        todos: state.todos
    }
};

const mapDispatchToProps = {
    addTodo
};

export default connect(mapStateToProps, mapDispatchToProps)(Todo);