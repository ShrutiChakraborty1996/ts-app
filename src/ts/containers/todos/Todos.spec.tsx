import * as React from "react";
import * as renderer from "react-test-renderer";
import {generateMockStore} from "../../mockStore";
import {Provider} from "react-redux";
import Todos from "./Todos";

describe('Todo connected Component', () => {
    const initialState: any = [];
    let mockStore = generateMockStore(initialState);

    beforeEach(() => {
        mockStore.clearActions();
    });

    describe('when todo is passed', () => {
        const mockState = {
            todos: [
                {title: "sample"}
            ]
        };
        mockStore = generateMockStore(mockState);

        it('should render Todo Component', () => {
            const component = renderer.create(<Provider store={mockStore}><Todos/></Provider>);
            expect(component).toMatchSnapshot();
        });
    });
});