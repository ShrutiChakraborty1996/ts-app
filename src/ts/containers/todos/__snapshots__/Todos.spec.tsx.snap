// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo connected Component when todo is passed should render Todo Component 1`] = `
<div
  className="todo-form"
>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
    />
    <button
      type="submit"
    >
      +
    </button>
  </form>
  <ul>
    <li>
      sample
    </li>
  </ul>
</div>
`;
